### DISCUSSION TITLE
Is there a way to make vite watch ignore changes to test files, but vitest watch only them?

### DISCUSSION THREAD

**Discussion by Grsz** [vitest-dev/vitest#6662] [Q&A]
I tried to define both `vitest.config`, and `vite.config`, but it seems like they don't "merge" which was a long shot to expect anyways, also it's recommended to use only `vite.config` when both vite and vitest are used.

When I set
```
    server: {
      watch: {
        ignored: ['**/coverage/**', '**/*.test.tsx'],
      },
    },
```
in `vite.config`, vitest stops listening to test file changes too.

Is there a way to solve this?

**Comment by sheremet-va** [comment]
You can define the value conditionally:

```
    server: {
      watch: {
        ignored: process.env.VITEST ? ['**/coverage/**'] : ['**/coverage/**', '**/*.test.tsx'],
      },
    },
```
